Add Non-Regex, Non-Lambda overloads for replace methods

nathanfranke opened the pull request:
Example usage:
---

Before:
```java
Component message = Component.text("Your nickname is now {}.", NamedTextColor.YELLOW)
        .replaceFirstText(Pattern.compile("{}", Pattern.LITERAL), old -> Component.text(nickname, NamedTextColor.GREEN).toBuilder());
```

After:
```java
Component message = Component.text("Your nickname is now {}.", NamedTextColor.YELLOW)
        .replaceFirstText("{}", Component.text(nickname, NamedTextColor.GREEN));
```

zml2008 commented:
This looks good -- but looks like you accidentally got kashike's commit into your history. Try doing an interactive rebase onto master to strip that out.

nathanfranke commented:
Fixed. I actually did that on purpose because this PR depended on kashike's

kashike commented on api/src/main/java/net/kyori/adventure/text/Component.java:
```diff
@@ -1468,6 +1468,18 @@ default boolean hasStyling() {
+  /**
+   * Finds and replaces text within any {@link Component}s using a string literal.
+   *
+   * @param old a string literal
```
Rename this parameter from `old` to `search`.

kashike commented on api/src/main/java/net/kyori/adventure/text/Component.java:
```diff
@@ -1480,6 +1492,18 @@ default boolean hasStyling() {
+  /**
+   * Finds and replaces the first occurrence of text within any {@link Component}s using a string literal.
+   *
+   * @param old a string literal
```
Rename this parameter from `old` to `search`.

nathanfranke commented:
I added a new overload for every existing `Pattern` overload.

Only problem is this code is repeated twice. `(index, replaced) -> replaced < numberOfReplacements ? PatternReplacementResult.REPLACE : PatternReplacementResult.STOP`

Tried to make a method but private methods aren't supported in interfaces until Java 9.

zml2008 commented on api/src/main/java/net/kyori/adventure/text/Component.java:
```diff
@@ -1505,6 +1542,19 @@ default boolean hasStyling() {
   }
 
+  /**
+   * Finds and replaces {@code n} instances of text within any {@link TextComponent}s using a string literal.
```
this should be updated to mention `fn`
